File: veloProweb-frontend/src/app/models/DTO/purchase-request-dto.ts
import { Supplier } from "../Entity/Purchase/supplier";
import { PurchaseDetailDTO } from "./purchase-detail-dto";

export interface PurchaseRequestDTO {
    id: number;
    date: string;
    supplierId: number;
    documentType: string;
    document: string;
    tax: number;
    total: number;
    detailList: PurchaseDetailDTO[];
}

File: veloProWeb/src/main/java/com/veloProWeb/model/dto/purchase/PurchaseRequestDTO.java
package com.veloProWeb.model.dto.purchase;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.List;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class PurchaseRequestDTO {
    private Long id;
    private LocalDate date;
    private Long idSupplier;
    private String documentType;
    private String document;
    private int tax;
    private int total;
    private List<DetailPurchaseDTO> detailList;
}
